### PR TITLE
sane-backend: deprecate phases

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#28910
ping @symphorien, @Ekleog, @peti 

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by peti — CHANGES_REQUESTED** [review]
This code is somewhat unidiomatic. I think it would be much easier to just drop the assignment of phases.

**Comment by Stunkymonkey** [comment]
if we drop `phases = [ ...` we would have to add at least the line `dontUnpack = true;`. maybe more.
Do you prefer this? I personally prefer it the current way of running only the commands.

**Comment by peti** [comment]
Yes, I would prefer to stay within the realm of stdenv.mkDerivation. This is what everyone knows and is familiar with.

**Comment by Stunkymonkey** [comment]
first of all you are the maintainer of this packet, so it should be how you want to manage it (using `stdenv.mkDerivation`).
Personally it bothers me a little that the use of `runCommand(Local)` would be perfect here. You really prefer using `stdenv.mkDerivation`?

**Review by Artturin — COMMENTED** [review]
before
```
nix-repl> mkSaneConfig { paths = [pkgs.sane-backends]; disabledDefaultBackends = ["v4l"]; }
«derivation /nix/store/v9p7cycmpa68d2ngmphd8v158qafk8b7-sane-config.drv»
```
and produces `out`

after
```
nix-repl> mkSaneConfig { paths = [pkgs.sane-backends]; disabledDefaultBackends = ["v4l"]; }
"/nix/store/z29m4sic7mx7m46y4xvlmmszg74lwb14-sane-configif [ -e \"/nix/store/34za0pidcxqdlz2kwz3dc4vbmva4wyh0-sane-backends-1.0.32/lib/sane\" ]; then\n  find \"/nix/store/34za0pidcxqdlz2kwz3dc4vbmva4wyh0-sane-backends-1.0.32/lib/sane\" -maxdepth 1 -not -type d | while read backend; do\n    symlink \"$backend\" \"$out/lib/sane/$(basename \"$backend\")\"\n  done\nfi\n\nif [ -e \"/nix/store/34za0pidcxqdlz2kwz3dc4vbmva4wyh0-sane-backends-1.0.32/etc/sane.d\" ]; then\n  find \"/nix/store/34za0pidcxqdlz2kwz3dc4vbmva4wyh0-sane-backends-1.0.32/etc/sane.d\" -maxdepth1 -not -type d | while read conf; do\n    name=\"$(basename $conf)\"\n    if [ \"$name\" = \"dll.conf\" ] || [ \"$name\" = \"saned.conf\" ] || [ \"$name\" = \"net.conf\" ]; then\n      cat \"$conf\" >> \"$out/etc/sane.d/$name\"\n    else\n      symlink \"$conf\" \"$out/etc/sane.d/$name\"\n    fi\n  done\nfi\n\nif [ -e \"/nix/store/34za0pidcxqdlz2kwz3dc4vbmva4wyh0-sane-backends-1.0.32/etc/sane.d/dll.d\" ]; then\n  find \"/nix/store/34za0pidcxqdlz2kwz3dc4vbmva4wyh0-sane-backends-1.0.32/etc/sane.d/dll.d\" -maxdepth 1 -not -type d | while read conf; do\n    symlink \"$conf\" \"$out/etc/sane.d/dll.d/$(basename $conf)\"\n  done\nfi\ngrep -q 'v4l' $out/etc/sane.d/dll.conf || { echo 'v4l is not a default plugin in $SANE_CONFIG_DIR/dll.conf'; exit 1; }\nsubstituteInPlace $out/etc/sane.d/dll.conf --replace 'v4l' '# v4l disabled in nixos config'\n"
```

maybe something to do with the way it is called

File: pkgs/applications/graphics/sane/config.nix [hunk 1]
```diff
@@ -35,7 +35,9 @@ installSanePath = path: ''
 in
 stdenv.mkDerivation {
   name = "sane-config";
-  phases = "installPhase";
+  dontUnpack = true;
+  dontPatch = true;
```
Comment (peti) [inline pkgs/applications/graphics/sane/config.nix:39]: Why are the patch and fixup phases disabled? There should be no harm in keeping those, no? Fixup in particular seems important to run.

Comment (Stunkymonkey) [reply pkgs/applications/graphics/sane/config.nix]: i thought, that everything should stay the same as before and disable them. But i am in favor of enabling them.

**Review by peti — APPROVED** [review]

**Pull request merged**